src/Carbon/Translator.php

```diff
@@ -361,7 +361,9 @@ public function setLocale($locale)
             parent::setLocale($macroLocale);
         }
 
-        if ($this->loadMessagesFromFile($locale) || $this->initializing) {
+        $loaded = $this->loadMessagesFromFile($locale);
```

Comment:
This code has been changed on purpose: `loadMessagesFromFile` method should always be called.